packages/playwright-test/src/transform.ts

```diff
@@ -135,39 +144,50 @@ export function transformHook(code: string, filename: string, isModule = false):
-    [require.resolve('@babel/plugin-proposal-logical-assignment-operators')],
-    [require.resolve('@babel/plugin-proposal-nullish-coalescing-operator')],
-    [require.resolve('@babel/plugin-proposal-optional-chaining')],
-    [require.resolve('@babel/plugin-syntax-json-strings')],
-    [require.resolve('@babel/plugin-syntax-optional-catch-binding')],
-    [require.resolve('@babel/plugin-syntax-async-generators')],
-    [require.resolve('@babel/plugin-syntax-object-rest-spread')],
-    [require.resolve('@babel/plugin-proposal-export-namespace-from')],
-  ] as any;
-
-  if (tsconfigData) {
-    plugins.push([require.resolve('babel-plugin-module-resolver'), {
-      root: ['./'],
-      alias: tsconfigData.alias,
-      // Silences warning 'Could not resovle ...' that we trigger because we resolve
-      // into 'foo/bar', and not 'foo/bar.ts'.
-      loglevel: 'silent',
-    }]);
+  const plugins = [];
+
+  if (isTypeScript) {
+    plugins.push(
+        [require.resolve('@babel/plugin-proposal-class-properties')],
+        [require.resolve('@babel/plugin-proposal-numeric-separator')],
+        [require.resolve('@babel/plugin-proposal-logical-assignment-operators')],
+        [require.resolve('@babel/plugin-proposal-nullish-coalescing-operator')],
+        [require.resolve('@babel/plugin-proposal-optional-chaining')],
+        [require.resolve('@babel/plugin-syntax-json-strings')],
+        [require.resolve('@babel/plugin-syntax-optional-catch-binding')],
+        [require.resolve('@babel/plugin-syntax-async-generators')],
+        [require.resolve('@babel/plugin-syntax-object-rest-spread')],
+        [require.resolve('@babel/plugin-proposal-export-namespace-from')]
+    );
+
+    if (tsconfigData) {
+      plugins.push([require.resolve('babel-plugin-module-resolver'), {
+        root: ['./'],
+        alias: tsconfigData.alias,
+        // Silences warning 'Could not resovle ...' that we trigger because we resolve
+        // into 'foo/bar', and not 'foo/bar.ts'.
+        loglevel: 'silent',
+      }]);
+    }
+
+    if (!isModule) {
+      plugins.push([require.resolve('@babel/plugin-transform-modules-commonjs')]);
+      plugins.push([require.resolve('@babel/plugin-proposal-dynamic-import')]);
+    }
+  } else {
+    // JavaScript file && PW_TEST_SOURCE_TRANSFORM was passed, so we apply it.
+
+    // Bail out on files outside of the preprocessor scope.
+    if (!externalPreprocessorScope || !filename.startsWith(externalPreprocessorScope))
+      return code;
   }
 
   if (process.env.PW_COMPONENT_TESTING)
     plugins.unshift([require.resolve('@babel/plugin-transform-react-jsx')]);
 
-  if (!isModule) {
-    plugins.push([require.resolve('@babel/plugin-transform-modules-commonjs')]);
-    plugins.push([require.resolve('@babel/plugin-proposal-dynamic-import')]);
-  }
 
-  if (process.env.PW_TEST_SOURCE_TRANSFORM)
-    plugins.push([process.env.PW_TEST_SOURCE_TRANSFORM]);
+  if (scriptPreprocessor)
```

Comment:
Only add this one when file is in scope? Otherwise we transform ts files outside the scope.

Reply:
Done!